udata/search/commands.py

```diff
@@ -69,150 +46,31 @@ def index_model(index_name, adapter, timeout=None):
-        'index': {
-            'refresh_interval': '-1'
-        }
-    }, request_timeout=timeout)
-
-
-def enable_refresh(index_name, timeout=None):
-    '''
-    Enable refresh and force merge. To be used after indexing.
-
-    See: https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-update-settings.html#bulk
-    '''  # noqa
-    timeout = timeout or current_app.config['ELASTICSEARCH_INDEX_TIMEOUT']
-    refresh_interval = current_app.config['ELASTICSEARCH_REFRESH_INTERVAL']
-    es.indices.put_settings(index=index_name, body={
-        'index': {'refresh_interval': refresh_interval}
-    }, request_timeout=timeout)
-    # Double the timeout as this can take some time
-    es.indices.forcemerge(index=index_name, request_timeout=timeout * 2)
-
-
-def set_alias(index_name, delete=True, timeout=None):
-    '''
-    Properly end an indexation by creating an alias.
-    Previous alias is deleted if needed.
-    '''
-    timeout = timeout or current_app.config['ELASTICSEARCH_INDEX_TIMEOUT']
-    log.info('Creating alias "%s" on index "%s"', es.index_name, index_name)
-    if es.indices.exists_alias(name=es.index_name):
-        alias = es.indices.get_alias(name=es.index_name)
-        previous_indices = alias.keys()
-        if index_name not in previous_indices:
-            es.indices.put_alias(index=index_name, name=es.index_name,
-                                 request_timeout=timeout)
-        for index in previous_indices:
-            if index != index_name:
-                es.indices.delete_alias(index=index, name=es.index_name,
-                                        request_timeout=timeout)
-                if delete:
-                    es.indices.delete(index=index, request_timeout=timeout)
-    else:
-        es.indices.put_alias(index=index_name, name=es.index_name,
-                                               request_timeout=timeout)
-
-
-@contextmanager
-def handle_error(index_name, keep=False, timeout=None):
-    '''
-    Handle errors while indexing.
-    In case of error, properly log it, remove the index and exit.
-    If `keep` is `True`, index is not deleted.
-    '''
-    # Handle keyboard interrupt
-    signal.signal(signal.SIGINT, signal.default_int_handler)
-    signal.signal(signal.SIGTERM, signal.default_int_handler)
-    has_error = False
-    try:
-        yield
-    except KeyboardInterrupt:
-        print('')  # Proper warning message under the "^C" display
-        log.warning('Interrupted by signal')
-        has_error = True
-    except Exception as e:
-        log.error(e)
-        has_error = True
-    if has_error:
-        if not keep:
-            log.info('Removing index %s', index_name)
-            es.indices.delete(index=index_name, request_timeout=timeout)
-        sys.exit(-1)
+    for doc in docs:
+        try:
+            produce(model, doc)
+        except Exception as e:
+            log.error('Unable to index %s "%s": %s', model, str(doc.id),
```

Comment:
this generate error when `udata init`
`AttributeError: 'dict' object has no attribute 'id'`